tests/core/pyspec/eth2spec/test/altair/block_processing/sync_aggregate/test_process_sync_aggregate.py

```diff
@@ -402,3 +407,189 @@ def test_proposer_in_committee_with_participation(spec, state):
+        sync_committee_bits=committee_bits,
+        sync_committee_signature=compute_aggregate_sync_committee_signature(
+            spec,
+            state,
+            block.slot - 1,
+            committee_indices,  # with exited validator removed
+        )
+    )
+    yield from run_sync_committee_processing(spec, state, block)
+
+
+@with_altair_and_later
+@spec_state_test
+@always_bls
+def test_sync_committee_with_participating_withdrawable_member(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for exit
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    # move forward via some blocks
+    for _ in range(3):
+        next_epoch_via_block(spec, state)
+
+    committee_indices = compute_committee_indices(spec, state)
+    rng = random.Random(1010)
+
+    exited_validator_index = rng.sample(committee_indices, 1)[0]
+    exits = prepare_signed_exits(spec, state, [exited_validator_index])
+    assert len(exits) == 1
+    voluntary_exit = exits.pop()
+    spec.process_voluntary_exit(state, voluntary_exit)
+
+    target_validator = state.validators[exited_validator_index]
+    target_validator.withdrawable_epoch = target_validator.exit_epoch + 1
+
+    target_slot = (target_validator.withdrawable_epoch + 1) * spec.SLOTS_PER_EPOCH
+    transition_to(spec, state, target_slot)
+
+    exited_validator_indices = get_unslashed_exited_validators(spec, state)
+    assert exited_validator_index in exited_validator_indices
+    exited_pubkey = state.validators[exited_validator_index].pubkey
+    assert exited_pubkey in state.current_sync_committee.pubkeys
+    current_epoch = spec.get_current_epoch(state)
+    assert current_epoch > state.validators[exited_validator_index].withdrawable_epoch
+
+    block = build_empty_block_for_next_slot(spec, state)
+    block.body.sync_aggregate = spec.SyncAggregate(
+        sync_committee_bits=[True] * len(committee_indices),
+        sync_committee_signature=compute_aggregate_sync_committee_signature(
+            spec,
+            state,
+            block.slot - 1,
+            committee_indices,  # full committee signs
+        )
+    )
+    yield from run_sync_committee_processing(spec, state, block)
+
+
+@with_altair_and_later
+@spec_state_test
+@always_bls
+def test_sync_committee_with_nonparticipating_withdrawable_member(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for exit
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    # move forward via some blocks
+    for _ in range(3):
+        next_epoch_via_block(spec, state)
+
+    committee_indices = compute_committee_indices(spec, state)
+    rng = random.Random(1010)
+
+    exited_validator_index = rng.sample(committee_indices, 1)[0]
+    exits = prepare_signed_exits(spec, state, [exited_validator_index])
+    assert len(exits) == 1
+    voluntary_exit = exits.pop()
+    spec.process_voluntary_exit(state, voluntary_exit)
```

Comment:
Why do the `process_voluntary_exit` and full 4-epoch transition to get to the exit vs just setting exit_epoch and withdrawable_epoch?

Reply:
I understand that it leads to a more likely correct situation but I'm not certain the overhead is worth it here, especially when you artificially reduce the `withdrawable_epoch` anyway

Reply:
that's true, was aiming for whatever is more "realistic" but can just manually set the exit and withdrawable epochs here

Reply:
refactored some of the duplicated code and moved to manually setting exit and withdrawable epochs